apps/extension/server/src/server.ts

```diff
@@ -259,6 +265,11 @@ async function handleDocumentChange(textDocument: TextDocument): Promise<void> {
       });
     }
   } catch (e) {
```

Comment:
This currently works differently for config-related errors (eg. when a transition targets a state that doesn't exist) when we have `tsTypes` and when we don't have them.

With `tsTypes` it crashes because `getTypegenData` call leads to `createIntrospectableMachine` and it crashes there or in some code that introspects that result. In such a case we nicely display the error in the Editor (with a caveat that you can introduce a config-related error to a different machine in a file and you'd still see the error in the Editor when displaying something else, this could have been fixed).

Without `tsTypes` the invalid config gets sent to the Editor and it crashes there in the `UPDATE_CONFIG` handler within the `fromMachineConfig` call. No error is displayed in this case.

It could be easier to patch this in the Editor but I feel like we should really fix this here. There is currently a no good way to check for those kinds of errors though. You could try to abuse the `createIntrospectableMachine` to check for this though.

What you'd have to do is to map those results differently:
https://github.com/statelyai/xstate-tools/blob/83d5b671b67f378366939312c3ca253510c6fb1b/apps/extension/server/src/server.ts#L199-L210

We need to end up with an array like this `Array<{ types?: TypegenData, configError?: string }>` and handle that appropriately in logic that follows this `.map`.